lib/Model/Importer/Binary/ELFImporter.cpp

```diff
@@ -142,6 +142,9 @@ class FilePortion2 {
   }
 };
 
+// Here are the header files that implement target specific importing.
+#include "MipsELFImporter.h"
+
```

Comment:
Keep the includes at the top.
Also, can you elaborate the goal of this refactoring? I expected to have a `virtual` class that defines methods to hook certain parts of the ELF parsing in a way that we basically never mention explicitly MIPS.
Here the situation improved but it seems that we're only half way there.

Or was your goal something else?

Reply:
I agree with this, and it will be addressed in the following update.